agent/src/main/java/com/teamscale/jacoco/agent/testimpact/CoverageToJsonStrategyBase.java

```diff
@@ -53,12 +54,12 @@ public CoverageToJsonStrategyBase(JacocoRuntimeController controller, AgentOptio
 	public List<PrioritizableTestCluster> testRunStart(List<ClusteredTestDetails> availableTests,
 													   boolean includeNonImpactedTests,
 													   boolean includeAddedTests, boolean includeFailedAndSkipped,
-													   String baseline) throws IOException {
+													   String baseline, String baselineRevision) throws IOException {
```

Comment:
Only changing it for the strategy classes probably doesn't make much sense. After refactoring, impacted places are:
* tests
* the CLI
* tia-client library via `com.teamscale.tia.client.ITestwiseCoverageAgentApi`. 

The latter might be the more interesting case to discuss.

Btw, all of this does not affect our plugins but only our internal data model and the tia client library. This is just a realization I had (maybe again). It's very confusing to me that we have features like retrieving impacted tests and uploading coverage duplicated in different places (_impacted tests retrieval_ in impacted test engine and the agent; _uploading coverage_ is a separate job that uses the Teamscale client in Gradle and in the maven plugin, the agent does the upload).

With that in mind, the thing I'm asking myself is: how do we want to design our Teamscale client API (Java) and how different should it be from the REST API? The REST API accepts both baseline parameters simultaneously and let's Teamscale decide. If we want to keep the Java API close to  the REST API as possible, I'd keep the code as is. Otherwise we could change it. I don't have a very strong opinion on this but am slightly tending towards keeping the API as close to the REST API as possible. What's your take? 

I'm also happy to discuss this in a short meeting. Then we can also have a closer look on what the impact would be of the refactoring.

Reply:
I'd leave it up to you what you think is the better implementation. We can ofc also have a look together, but I don't have strong opinion. Was just an idea.

Reply:
I'll have another look.

Reply:
First off, thanks for the suggestion. Bit late, but forgot before. I generally like the idea :)

Also, I don't understand my last comment anymore xD. I don't know, how I thought that the CLI and the ITestwiseCoverageAgentApi were impacted. AFAICT now, we're only piping the data from the Http request that arrives from outside to the agent to Teamscale here, so making it explicit that only timestamp or revision needs to bet set only adds some if statement when creating the baseline object only to use both, timestamp and revision, in the request to Teamscale (or we again add some if statements). So I'll leave it as is. 

The more interesting places would probably be impacted test engine and the Gradle and Maven plugins. And we could not only do this for the baseline, but also for the (end)Commit. In the agent module, a fitting class already exists: `com.teamscale.jacoco.agent.commit_resolution.git_properties.CommitInfo` (has a String revision and CommitDescriptor commit). Maybe would could move this to the Teamscale client and use it everywhere?!
Either way, that sounds to me like a larger refactoring => Maintenance ticket.